### PR TITLE
iox-#609 Add generic QNX 7.0 toolchain files

### LINKED ISSUE #609: Add generic QNX toolchain files
As mentioned in the `installation.md` the easiest way to build iceoryx on QNX is to use the build script and providing a toolchain file.

Question is if we want to add some kind of generic toolchain file(s) for QNX to get people started.  See example file in associated PR.

Edit 2021-03-19:  In yesterday's dev meeting ([minutes](https://github.com/eclipse-iceoryx/iceoryx/wiki/2021-03-18-Eclipse-iceoryx-developer-meetup)) it was agreed on adding these toolchain files.

### PR DESCRIPTION
**Pull request by marthtz**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/advanced/best-practice-for-testing.md

## Notes for Reviewer
Add generic toolchain files to be used for QNX SDP 7.0 builds on ARM and X86_64.

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Closes #609

### CODE REVIEW

**Review by elfenpiff — DISMISSED** [review]

**Review by elBoberido — APPROVED** [review]

File: tools/toolchains/qnx/qnx_sdp70_common.cmake [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
...butes;cxx_auto_type;cxx_constexpr;cxx_decltype;cxx_decltype_incomplete_return_types;cxx_default_function_template_args;cxx_defaulted_functions;cxx_defaulted_move_initializers;cxx_delegating_constructors;cxx_deleted_functions;cxx_enum_forward_declarations;cxx_explicit_conversions;cxx_extended_friend_declarations;cxx_extern_templates;cxx_final;cxx_func_identifier;cxx_generalized_initializers;cxx_inheriting_constructors;cxx_inline_namespaces;cxx_lambdas;cxx_local_type_template_args;cxx_long_long_type;cxx_noexcept;cxx_nonstatic_member_init;cxx_nullptr;cxx_override;cxx_range_for;cxx_raw_string_literals;cxx_reference_qualified_functions;cxx_right_angle_brackets;cxx_rvalue_references;cxx_sizeof_member;cxx_static_assert;cxx_strong_enums;cxx_thread_local;cxx_trailing_return_types;cxx_unicode_literals;cxx_uniform_initialization;cxx_unrestricted_unions;cxx_user_literals;cxx_variadic_macros;cxx_variadic_templates;cxx_std_14;cxx_aggregate_default_initializers;cxx_attribute_deprecated;cxx_binary_literals;cxx_contextual_conversions;cxx_decltype_auto;cxx_digit_separators;cxx_generic_lambdas;cxx_lambda_init_captures;cxx_relaxed_constexpr;cxx_return_type_deduction;cxx_variable_templates;cxx_std_17")
```
Comment (mossmaurice) [inline tools/toolchains/qnx/qnx_sdp70_common.cmake:57]: `cxx_std_17` I don't know what this flag exactly causes, but it seems we might omit it.

**Review by mossmaurice — COMMENTED** [review]
LGTM, nice preparation for our QNX CI build

File: tools/toolchains/qnx/qnx_sdp70_common.cmake (hunk 1 above)
Comment (marthtz) [reply tools/toolchains/qnx/qnx_sdp70_common.cmake:57]: You're probably right.
We don't want "_Compiler mode is at least C++ 17._", yet.

**Comment by mossmaurice** [comment]
@marthtz What do you think about making them the default toolchain files in `./tools/iceoryx_build_test.sh` script? I'd like to make it as easy as possible for folks who want to build on QNX.

**Comment by marthtz** [comment]
>  I'd like to make it as easy as possible for folks who want to build on QNX.

I updated the QNX section in `installation.md` with the generic toolchain files.

**Review by elBoberido — APPROVED** [review]

**Review by dkroenke — APPROVED** [review]

**Pull request merged**